lib/smart_proxy_ansible/variables_extractor.rb

```diff
@@ -6,26 +8,11 @@ class << self
         def extract_variables(role_path)
           role_files = Dir.glob("#{role_path}/defaults/**/*.yml") +
                        Dir.glob("#{role_path}/defaults/**/*.yaml")
-          # not anything matching item, }}, {{, ansible_hostname or 'if'
-          variables = role_files.map do |role_file|
-            candidates = File.read(role_file)
-                         .scan(/{{(.*?)}}/).select do |param|
-              param.first.scan(/item/) == [] && param.first.scan(/if/) == []
-            end.flatten
-            # Sometimes inside the {{ }} there's a OR condition. In such a case,
-            # let's split and choose possible variables (variables cannot
-            # contain parenthesis)
-
-            candidates.map do |variable|
-              variable.split('|').map(&:strip).select do |var|
-                !var.include?('(') && # variables are not parenthesis
-                !var.include?('[') && # variables are not arrays
-                !var.include?('.') && # variables are not objects
-                !var.include?("'") # variables are not plain strings
-              end
-            end unless candidates.nil?
-          end.compact.flatten.uniq.map(&:strip)
-          variables
+          role_files.flat_map do |role_file|
+            loaded_yaml = YAML.load_file(role_file) || {}
```

Comment:
this can raise e.g. Psych::SyntaxError if it's not valid YAML, should we rescue and raise ReadVariablesException in case this line raises anything?